tests/json/crud/options.rs

```diff
@@ -102,32 +98,26 @@ impl FromJson for FindOneAndUpdateOptions {
     fn from_json(object: &Object) -> FindOneAndUpdateOptions {
         let mut options = FindOneAndUpdateOptions::new();
 
-        let f = |x| Some(Bson::from_json(x));
-        options.projection = match object.get("projection").and_then(f) {
-            Some(Bson::Document(doc)) => Some(doc),
-            _ => None,
-        };
+        if let Some(Bson::Document(projection)) = object.get("projection").map(Bson::from_json) {
```

Comment:
I'll be honest; I have absolutely no idea why I implemented this (and the similar cases below) in such a convoluted way originally. I'll just have to chalk it up to my lack of familiarity with Rust at that point.